packages/cozy-intent/src/view/components/WebviewIntentProvider.tsx

```diff
@@ -71,11 +72,12 @@ const isValidEnv = (): boolean => {
 
 export const WebviewIntentProvider = ({
   children,
+  setBarContext,
   webviewService
 }: Props): ReactElement => {
   const [connection, setConnection] = useState<WebviewConnection>()
   const [service, setService] = useState<WebviewService | void>(webviewService)
-  const setBarWebviewContext = getBarInitAPI()
+  const setBarWebviewContext = setBarContext || getBarInitAPI()
```

Comment:
I will approuve once this test is added.
```
  it('should allow cozy-bar v8 to inject callback', async () => {
    const mockSetBarContext = jest.fn()
    const { findByText } = render(
      <WebviewIntentProvider setBarContext={mockSetBarContext}>Hello</WebviewIntentProvider>
    )

    expect(await findByText('Hello')).toBeTruthy()
    expect(mockSetBarContext).toBeCalledWith(expect.any(WebviewService))
    expect(mockSetBarContext).toBeCalledTimes(1)
  })
  ```

Reply:
added